oommfc/drivers/hysteresisdriver.py

```diff
@@ -59,13 +59,31 @@ class HysteresisDriver(Driver):
     ]
 
     def _checkargs(self, **kwargs):
-        Hmin, Hmax, n = kwargs["Hmin"], kwargs["Hmax"], kwargs["n"]
-        for i in [Hmin, Hmax]:
+        # check the default arguments for a symmetric hysteresis loop
+        if any(item in kwargs for item in ["Hmin", "Hmax", "n"]) and "Hsteps" in kwargs:
+            msg = "Cannot define both (Hmin, Hmax, n) and Hsteps."
+            raise ValueError(msg)
+
+        if all(item in kwargs for item in ["Hmin", "Hmax", "n"]):
+            self._checkvalues(kwargs["Hmin"], kwargs["Hmax"], kwargs["n"])
+        elif "Hsteps" in kwargs:
+            for Hstart, Hend, n in kwargs["Hsteps"]:
+                self._checkvalues(Hstart, Hend, n)
```

Comment:
The checks have several implicit assumptions that should be removed:

1. We should check if `kwargs["Hstep"]` is actually iterable before iterating over it.
2. For each element in `kwargs["Hstep"]` we should first check that it contains three elements before unpacking those.

Reply:
Please look up the suggested concept of verification in the latest commit.

Reply:
Changes look overall good. I added one additional check, see comment below.